### PR TITLE
chore(test): remove the karma / browserstack test infrastructure

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
We no longer want to run the workflow nor do we want this code hanging around! Now that #5458 is closed we don't need to run the karma tests anymore.

STENCIL-1239

## What is the current behavior?

Although we've removed all of the Karma tests we still have the Karma / Browerstack infrastructure hanging around. Unfortunately it's still causing build failures, such as this one: https://github.com/ionic-team/stencil/actions/runs/8466788671/job/23196360326?pr=5598

So we should remove it ASAP.

## What is the new behavior?

No more Karma! No more browserstack!

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

If the build passes I believe we're good.

In order to land this PR though we'll need to remove the required check for the browserstack gh actions workflow. We should also then delete the related secret.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -256,20 +256,6 @@
...
-      "args": [
-        "${workspaceFolder}/bin/stencil",
-        "build",
-        "--max-workers=0",
-        "--dev",
-        "--config",
-        "${workspaceFolder}/test/karma/stencil.config.ts"
-      ],
-      "protocol": "inspector"
-    },
```
Comment (rwaskiewicz) [inline .vscode/launch.json:272]: I think we should rm the configuration after this deleted one as well - the Karma Prerender test has moved as well

File: package.json [hunk 2]
```diff
@@ -46,11 +46,9 @@
     "test.dist": "node scripts/build --validate-build",
     "test.end-to-end": "cd test/end-to-end && npm ci && npm test && npm run test.dist",
     "test.jest": "node --experimental-vm-modules ./node_modules/jest/bin/jest.js",
-    "test.karma": "cd test/karma && npm ci && npm run karma",
-    "test.karma.prod": "cd test/karma && npm ci && npm run karma.prod",
     "test.wdio": "cd test/wdio && npm ci && npm run test",
     "test.wdio.testOnly": "cd test/wdio && npm ci && npm run wdio",
-    "test.prod": "npm run test.dist && npm run test.end-to-end && npm run test.jest && npm run test.karma && npm run test.sys.node && npm run test.testing && npm run test.analysis",
+    "test.prod": "npm run test.dist && npm run test.end-to-end && npm run test.jest && npm run test.sys.node && npm run test.testing && npm run test.analysis",
```
Comment (rwaskiewicz) [inline package.json:51]: Can we add `test.wdio` to this script?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (alicewriteswrongs) [reply .vscode/launch.json:272]: indeed we should! missed that one

File: test/karma/README.md [hunk 3]
```diff
@@ -1,116 +0,0 @@
-# Karma Testing
```
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply test/karma/README.md:1]: I'll do that in a follow-up PR

File: package.json (hunk 2 above)
Comment (alicewriteswrongs) [reply package.json:51]: sure, makes sense

Comment (alicewriteswrongs) [reply package.json]: I don't know if we use this script anywhere though! I kinda forgot it existed

File: .github/workflows/README.md [hunk 4]
```diff
@@ -117,18 +117,7 @@ When a `git push` is made to a branch, Stencil's CI is designed to stop existing
 branch.
 A new CI run (of each workflow) will begin upon stopping the existing job(s) using the new `HEAD` of the branch.
 
-### BrowserStack (`browserstack.yml`)
```
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply .github/workflows/README.md:120]: I'll do a follow-up PR for that and the other README change

File: test/karma/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply test/karma/README.md:1]: Oh, I do (locally) 😆

**Review by rwaskiewicz — APPROVED** [review]

**Review by christian-bromann — APPROVED** [review]
🔪

**Pull request merged**